clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -466,8 +468,17 @@ CIRGenModule::createCIRFunction(mlir::Location loc, StringRef name,
   {
     mlir::OpBuilder::InsertionGuard guard(builder);
 
+    // Some global emissions are triggered while emitting a function, e.g.
+    // void s() { x.method() }
+    //
+    // Be sure to insert a new function before a current one.
+    if (curCGF)
```

Comment:
I notice the incubator is storing this to a local variable. Is there a possibility that it may be changed before we use it on line 480?

Reply:
I don't see a clear code path that could change `curCGF`, but in case I missed something I updated the code and now it follows the incubator approach.